buildSrc/src/test/kotlin/com/example/model/WeirdCombo.kt

```diff
@@ -0,0 +1,335 @@
+                    } catch (e: IllegalStateException) {
+                        throw JsonParseException(
+                            "Unable to parse json into type Fish",
+                            e
+                        )
+                    } catch (e: NumberFormatException) {
+                        throw JsonParseException(
+                            "Unable to parse json into type Fish",
+                            e
+                        )
+                    } catch (e: NullPointerException) {
+                        throw JsonParseException(
+                            "Unable to parse json into type Fish",
+                            e
+                        )
+                    }
+                }
+            }
+        }
+
+        public data class Bird(
+            public val food: Food,
+            public val canFly: Boolean,
+        ) : Anything() {
+            override fun toJson(): JsonElement {
+                val json = JsonObject()
+                json.add("food", food.toJson())
+                json.addProperty("can_fly", canFly)
+                return json
+            }
+
+            public companion object {
+                @JvmStatic
+                @Throws(JsonParseException::class)
+                public fun fromJson(jsonString: String): Bird {
+                    try {
+                        val jsonObject = JsonParser.parseString(jsonString).asJsonObject
+                        return fromJsonObject(jsonObject)
+                    } catch (e: IllegalStateException) {
+                        throw JsonParseException(
+                            "Unable to parse json into type Bird",
+                            e
+                        )
+                    }
+                }
+
+                @JvmStatic
+                @Throws(JsonParseException::class)
+                public fun fromJsonObject(jsonObject: JsonObject): Bird {
+                    try {
+                        val food = Food.fromJson(jsonObject.get("food").asString)
+                        val canFly = jsonObject.get("can_fly").asBoolean
+                        return Bird(food, canFly)
+                    } catch (e: IllegalStateException) {
+                        throw JsonParseException(
+                            "Unable to parse json into type Bird",
+                            e
+                        )
+                    } catch (e: NumberFormatException) {
+                        throw JsonParseException(
+                            "Unable to parse json into type Bird",
+                            e
+                        )
+                    } catch (e: NullPointerException) {
+                        throw JsonParseException(
+                            "Unable to parse json into type Bird",
+                            e
+                        )
+                    }
+                }
+            }
+        }
+
+        public data class Paper(
```

Comment:
shouldn't we generate `sealed` classes here instead ?

Reply:
The `Anything` class is already sealed

Reply:
my bad didn't spot that. that's perfect. Nice job here !!